** (exit) exited in: GenServer.call(TwitchSniper.GiveawayManager, {:cycle}, 5000
)
    ** (EXIT) an exception was raised:
        ** (ArgumentError) argument error
            :erlang.apply([%TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}], :product, [])
            (streamingutils) TwitchSniper.GiveawayManager.handle_call/3
            (stdlib) gen_server.erl:629: :gen_server.try_handle_call/4
            (stdlib) gen_server.erl:661: :gen_server.handle_msg/5
            (stdlib) proc_lib.erl:239: :proc_lib.init_p_do_apply/3
[error] GenServer TwitchSniper.GiveawayManager terminating
** (ArgumentError) argument error
    :erlang.apply([%TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}], :product, [])
    (streamingutils) TwitchSniper.GiveawayManager.handle_call/3
    (stdlib) gen_server.erl:629: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:661: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:239: :proc_lib.init_p_do_apply/3
Last message: {:cycle}
State: {[%TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}, %TwitchSniper.Giveaway{delay: 200, organiser: "Idiot", product: "Some neet stuff"}], [%TwitchSniper.Giveaway{delay: 300000, organiser: "Idiot", product: "Some neet stuff"}]}
    (elixir) lib/gen_server.ex:564: GenServer.call/3
iex(12)>
